### DISCUSSION TITLE
Problem Flashing Pico via Windows USB

### DISCUSSION THREAD

**Discussion by Aarkain** [russhughes/st7789_mpy#155] [Q&A] [answered]
I'm planning to use an ST7735 with a Pico in a project, and I saw [here](https://forum.micropython.org/viewtopic.php?t=11087) that functionality for that chip had been added to this project. I seem to be having a problem getting the Pico to take the firmware, though. Normally I just use vanilla Micropython loaded through Thonny, but unfortunately Thonny doesn't let you select a local file to use (or if it does I have missed how.) What I am seeing online says I should be able to just drag and drop the UF2 file onto the Pico when it is mounted. I've confirmed that I **can** do that with the official Micropython UF2, but not with the firmware for RP2 from here. I admit, I am probably missing some major point or some obvious detail, but I am stumped. Any help or guidance would be greatly appreciated.

**Comment by ExperiMentor** [comment]
Please clarify whether you mean the flashing process does not work when you
drag and drop (what is the error message?), or that the flashing seems OK
but the program does not run as expected (what exactly happens?).

Have you tried copy & paste rather than drag and drop?

**Reply by Aarkain** [reply]
My apologies. The actual drag-and-drop process works, the file appears on the device but it is no different than adding any other file. After restarting the Pico it runs using the same firmware as before.

**Reply by russhughes** [reply]
That is definitely a problem. I will look at it today.

**Reply by russhughes** [reply]
I was able to download the firmware.uf2 from the repo and drag and drop it onto the Pico using windows 11. It took the new firmware.uf2 file:

![Screenshot 2024-02-06 174758](https://github.com/russhughes/st7789_mpy/assets/23587382/112b1162-fb10-4b7d-8fda-8a7fae2fb58e)

I tested several different firmware.uf2 file to verify that each time the new firmware was running. 

You might try using the flash_nuke.uf2 from 
https://learn.adafruit.com/getting-started-with-raspberry-pi-pico-circuitpython/circuitpython to clear the flash.

**Reply by Aarkain** [reply]
Thank you for looking into it. I'm on Windows 10 but I don't think that should matter. I had considered nuking the flash but was paranoid about bricking one of my boards. But then, it isn't like I haven't done it before by plugging a wire into the wrong row on a breadboard. I have been kind of amazed at how much abuse those little things can take. But anyway, I'll try it on a different PC first, which I should have done before I even asked the first question, and if that doesn't work I'll try nuking it. Thanks again for checking. Sorry to have wasted your time.

**Reply by Aarkain** [reply, marked as answer]
Just thought I'd pop over and let you know how this turned out. I finally got back to it and realized that the problem wasn't with your file or with my Pico, it was with me being "github illiterate". I "downloaded" the file the wrong way, and ended up with an unusable file that LOOKED like the right thing. Once I grokked what I had done, I downloaded it the right way and it flashed first time. So thanks again for the time you spent looking into it, and sorry for the wild goose chase.
